### PR TITLE
Explain how to install cert-manager using Flux

### PR DESCRIPTION
**Pull request by wallrj**
**Preview**: https://deploy-preview-1338--cert-manager-website.netlify.app/docs/installation/helm/#install-the-helm-chart-using-flux

I wanted to see for myself whether cert-manager can be installed using Flux and it seems to be quite straight forward,
so I documented it here.

I used some of the settings -- such as `--crds CreateReplace` -- that @slam-wowx had used in: https://github.com/cert-manager/cert-manager/issues/6409.

xref: 
 * #320 
 * https://github.com/cert-manager/cert-manager/issues/2197
 * https://github.com/cert-manager/cert-manager/issues/2380

### CODE REVIEW

File: content/docs/installation/helm.md [hunk 1]
```diff
@@ -282,7 +282,100 @@ Generally we recommend:
...
 [raise an issue](https://github.com/cert-manager/cert-manager/issues) to
 discuss.
 
+
+## Install the Helm chart using Flux
+
+The cert-manager Helm chart can be installed by [Flux](https://fluxcd.io/).
+
+First create a [`HelmRepository` resource](https://fluxcd.io/flux/components/source/helmrepositories/),
+configured with URL of the  cert-manager Helm repository.
```
Comment (jsoref) [inline content/docs/installation/helm.md:295]:
```suggestion
configured with URL of the cert-manager Helm repository.
```

**Review by jsoref — COMMENTED** [review]
/lgtm

File: content/docs/installation/helm.md [hunk 2]
```diff
@@ -282,7 +282,100 @@ Generally we recommend:
...
+```bash
+flux create helmrelease cert-manager \
+  --chart cert-manager \
+  --source HelmRepository/cert-manager.flux-system \
+  --release-name cert-manager \
+  --target-namespace cert-manager \
+  --create-target-namespace \
+  --crds CreateReplace \
+  --values values.yaml \
+  --chart-version '>1.12.0 <1.13.0'
```
Comment (erikgb) [inline content/docs/installation/helm.md:341]:
```suggestion
  --chart-version '1.12.x'
```

File: content/docs/installation/helm.md [hunk 3]
```diff
@@ -282,7 +282,100 @@ Generally we recommend:
...
+```bash
+flux create helmrelease cert-manager \
+  --chart cert-manager \
+  --source HelmRepository/cert-manager.flux-system \
+  --release-name cert-manager \
+  --target-namespace cert-manager \
+  --create-target-namespace \
+  --crds CreateReplace \
+  --values values.yaml \
+  --chart-version '>1.12.0 <1.14.0'
```
Comment (erikgb) [inline content/docs/installation/helm.md:358]:
```suggestion
  --chart-version '1.13.x'
```

**Review by erikgb — COMMENTED** [review]
Added a couple of suggestions, but LGTM at it stands.

/lgtm

File: content/docs/installation/helm.md [hunk 4]
```diff
@@ -282,7 +282,100 @@ Generally we recommend:
 You may want to consider your approach along with other tools that may offer
 helm compatible installs, for a standardized approach to managing CRD
 resources. If you have an approach that cert-manager does not currently
-support, then please 
+support, then please
 [raise an issue](https://github.com/cert-manager/cert-manager/issues) to
 discuss.
 
+
+## Install the Helm chart using Flux
```
Comment (inteon) [inline content/docs/installation/helm.md:290]: Shouldn't we add a separate Flux section?

Comment (inteon) [reply content/docs/installation/helm.md:290]: When I go to cert-manager.io, I probably would look for a Flux section if that is the tool I'm using.

Comment (wallrj) [reply content/docs/installation/helm.md]: What precisely do you propose?

I did consider adding a `flux.md` page, but decided that we'd end up with too many submenu items if we also add `argo.md`, `anthos.md`, `octopus.md` etc.

I also considered that Flux can install cert-manager in other ways too, so it seems appropriate to document the Flux Helm Controller here and in future document the Flex Kustomize controller in the kubectl applyt page.

I looked for examples from other projects and I found that Kyverno have a https://kyverno.io/docs/installation/platform-notes/ page where they include notes for ArgoCD users.
Couldn't find any other examples of projects that have a specific page for installing using flux or argo.

I have tried creating a "Continuous Deployment" page and move this tutorial there.
See what you think.

Comment (inteon) [reply content/docs/installation/helm.md]: What do you think about the name "GitOps"?

**Comment by hawksight** [comment]
I just ran through the flux installation and it worked perfectly. Thanks @wallrj - a good little intro to fluxcd for me.

/lgtm

Just a question for anyone who knows, did `fluxctl` become `flux` instead at some point? That tripped me up as I thought I had the CLI but turns out I did not.

**Review by hawksight — APPROVED** [review]
I just ran through the flux installation and it worked perfectly. Thanks @wallrj - a good little intro to fluxcd for me.

Just a question for anyone who knows, did `fluxctl` become `flux` instead at some point? That tripped me up as I thought I had the CLI but turns out I did not.

/lgtm

**Comment by erikgb** [comment]
> Just a question for anyone who knows, did `fluxctl` become `flux` instead at some point? That tripped me up as I thought I had the CLI but turns out I did not.

I think `fluxctl` was the CLI tool for the deprecated Flux v1 - which became `flux` in the current Flux v2.

File: content/docs/installation/helm.md (hunk 4 above)
Comment (wallrj) [reply content/docs/installation/helm.md]: I don't think it's appropriate. 

GitOps is about storing your infrastructure configuration in a Git repo and then using a continuous delivery / continuous deployment like Flux tool to apply it.
So this page is about the CD part. 

Why do you suggest it?

Comment (erikgb) [reply content/docs/installation/helm.md]: Also GitOps is the brand name Red Hat uses on their [ArgoCD hybrid](https://www.redhat.com/en/technologies/cloud-computing/openshift/gitops), so I would stay FAR away from that term. 😉

**Pull request merged**